gui/layers/_markers_layer.py

```diff
@@ -71,15 +77,36 @@ def __init__(
         self._coords = coords
 
         # Save the marker style params
+        size_array = np.array(size)
+        if len(size_array.shape) == 2:
+            assert(size_array.shape == self._coords.shape), \
```

Comment:
Would like to see this (and all other uses of `assert`) replaced with an if-statement + exception.

Reply:
Oh yeah I agree here, this should raise a `ValueError`, and same elsewhere. Happy to wait for another PR though.